src/main/java/core/basesyntax/Car.java

```diff
@@ -1,5 +1,115 @@
-package core.basesyntax;
-
-public class Car {
-
-}
+package core.basesyntax;
+
+public class Car implements Cloneable {
+    private String name;
+    private String color;
+    private int weight;
+    private int maxSpeed;
+    private int seats;
+    private Engine engine;
+
+    public Car(String name, String color, int weight, int maxSpeed, int seats) {
+        this.name = name;
+        this.color = color;
+        this.weight = weight;
+        this.maxSpeed = maxSpeed;
+        this.seats = seats;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public String getColor() {
+        return color;
+    }
+
+    public void setColor(String color) {
+        this.color = color;
+    }
+
+    public int getWeight() {
+        return weight;
+    }
+
+    public void setWeight(int weight) {
+        this.weight = weight;
+    }
+
+    public int getMaxSpeed() {
+        return maxSpeed;
+    }
+
+    public void setMaxSpeed(int maxSpeed) {
+        this.maxSpeed = maxSpeed;
+    }
+
+    public int getSeats() {
+        return seats;
+    }
+
+    public void setSeats(int seats) {
+        this.seats = seats;
+    }
+
+    public void setEngine(int cilindersNumber,
+                          int volumeCilinders,
+                          int weightEngine,
+                          String engineType,
+                          String kppType) {
+        this.engine = new Engine(cilindersNumber,
+                volumeCilinders,
+                weightEngine,
+                engineType,
+                kppType);
+    }
+
+    public void startEngine() {
+        if (engine != null) {
+            engine.startEngine();
+        } else {
+            System.out.println("Engine doesn't exist!");
+        }
+    }
+
+    public void stopEngine() {
+        if (engine != null) {
+            engine.stopEngine();
+        } else {
+            System.out.println("Engine doesn't exist!");
+        }
+    }
+
+    @Override
+    public Object clone() throws CloneNotSupportedException {
+        Car clonedCar = new Car(
+                this.getName(),
+                this.getColor(),
+                this.getWeight(),
+                this.getMaxSpeed(),
+                this.getSeats());
+        if (engine != null) {
+            clonedCar.engine = (Engine) this.engine.clone();

```

Comment:
do we need to use `this` and how we can omit casting?

Reply:
We have to check engine != null because we can create Car without an engine, and then if we call the clone() on car without engine we cath NullPointerException. Cast (Engine) I have fixed.